python/ql/src/semmle/python/regex.qll

```diff
@@ -124,16 +124,48 @@ abstract class RegexString extends Expr {
         )
     }
 
+    /** Escaped characters without any special handling (yet) */
+    private predicate singleEscape(int i) {
+        exists(string c |
+          c = this.getChar(i) and
+          c != "x" and c != "u" and c != "U" and c != "N"
+        )
+    }
+
+    /** Named unicode characters, eg \N{degree sign} */
+    private predicate escapedName(int start, int end) {
```

Comment:
As with `singleEscape`, this would apply also to, say, the string `"FooN{hello}"` where no escape appears. I think this should mention `this.escapingChar(start)` as well.

Reply:
I thought of it as local to the character predicate, is there a way to do that? However, it does make sense to have it on its own, so I have added `this.escapingChar(start)`. Now we could also add tests for it..